pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,640 @@
+        Note that even though information about ``Y[0]`` is contained in ``nums_frequency`` ,
+        ``ids`` determines which reconstructions are performed.
+
+        >>> with qml.Tracker(circuit.device) as tracker:
+        ...     rec = qml.fourier.reconstruct(circuit, {"Y": [(1,)]}, nums_frequency)(x, Y)
+        >>> tracker.totals
+        {'executions': 13}
+
+        As expected, we required :math:`2R+1=2\cdot 6+1=13` circuit executions. However, not
+        all frequencies below :math:`f_\text{max}=6` are present in the circuit, so that
+        a reconstruction using knowledge of the full frequency spectrum will be cheaper:
+
+        >>> spectra = {"Y": {(1,): [0., 1., 4., 5., 6.]}}
+        >>> with tracker:
+        ...     rec = qml.fourier.reconstruct(circuit, {"Y": [(1,)]}, None, spectra)(x, Y)
+        >>> tracker.totals
+        {'executions': 9}
+
+        We again obtain the full univariate dependence on ``Y[1]`` but with considerably
+        fewer executions on the quantum device.
+        Once we obtained the classical function that describes the dependence, no
+        additional circuit evaluations are performed:
+
+        >>> with tracker:
+        ...     for Y1 in np.arange(-np.pi, np.pi, 20):
+        ...         rec["Y"][(1,)](-2.1)
+        >>> tracker.totals
+        {}
+
+        If we want to reconstruct the dependence of ``circuit`` on ``x`` , we cannot use
+        ``nums_frequency`` if ``f`` is not an integer. One could rescale ``x`` to obtain
+        the frequency :math:`1` again, or directly use ``spectra`` . We will combine the
+        latter with another reconstruction with respect to ``Y[0]`` :
+
+        >>> spectra = {"x": {(): [0., f]}, "Y": {(0,): [0., 1.]}}
+        >>> with tracker:
+        ...     rec = qml.fourier.reconstruct(circuit, None, None, spectra)(x, Y, f=f)
+        >>> tracker.totals
+        {'executions': 5}
+        >>> recon_x = rec["x"][()]
+        >>> np.isclose(recon_x(x+0.5), circuit(x+0.5, Y, f=f)
+        True
+
+        Note that by convention, the parameter index for a scalar variable is ``()`` and
+        that the frequency :math:`0` always needs to be included in the spectra.
+        Furthermore, we here skipped the input ``ids`` so that the reconstruction
+        was performed for all keys in ``spectra`` .
+        The reconstruction with a single non-zero frequency
+        costs three evaluations of ``circuit`` for each, ``x`` and ``Y[0]`` . Performing
+        both reconstructions at the same position allowed us to save one of the
+        evaluations and reduce the number of calls to :math:`5`.
+
+        All examples above used that we already knew the frequency spectra of the
+        QNode of interest. However, this is in general not the case and we may need
+        to compute the spectrum first. This can be done with
+        :func:`.fourier.qnode_spectrum` :
+
+        >>> spectra = qml.fourier.qnode_spectrum(circuit)(x, Y)
+        >>> spectra.keys()
```

Comment:
@dwierichs the entire docstring looks 💯, much improved and very nice to read!

My only major suggestion is to move this paragraph to the end of the minimal example (above `.. UsageDetails::`), as I think a casual reader might 'lose sight' of it, and I think it is quite important that this link to `qnode_spectrum` is visible.

Reply:
You could even stick this paragraph inside a `.. seealso::`.